### ISSUE TITLE
CUDA OOM when accelerator.load_state() with multi-GPU training

### ISSUE THREAD

**Issue by rmokady** [huggingface/accelerate#1471]
Hi,

My code save the state using accelerate every few iterations
```
if accelerator.is_main_process:
    accelerator.save_state(save_path)
```

Then resuming the training conducted using
`accelerator.load_state(os.path.join(args.output_dir, path))`

This works well when using a single GPU, but fail for multi-GPU training

This issue #https://github.com/huggingface/accelerate/issues/296 suggest using `map_location="cpu"` (see also #https://github.com/huggingface/accelerate/commit/1e0b96f814dd248e4fb2311d721ba1a2ecb17496), but `accelerator.load_state(os.path.join(args.output_dir, path), map_location="cpu")` fail as well

Basically my code is using diffusers similar to [this](https://github.com/huggingface/diffusers/blob/main/examples/text_to_image/train_text_to_image.py)

Any help will be appreciated :)

**Comment by rmokady** [comment]
Solved by #https://github.com/huggingface/accelerate/commit/70920895e80f78d96d8f91e0beeb3ebdb8e5e5d6
